src/components/MasterDetails.js

```diff
@@ -137,6 +137,31 @@ export default class MasterDetails extends React.Component<MasterDetailsDefaultP
   }
   state: MasterDetailsState;
 
+  componentWillReceiveProps(newProps: MasterDetailsProps) {
+    const isRowPresent = newProps.rows.find((row) => {
```

Comment:
(As a short cut, I think you can see if the props.rows array is empty and then just set selectedRows to [] and detailsRowId to '' and be done with it...)